### ISSUE TITLE
Add catalog tail resource

### ISSUE THREAD

**Issue by agr** [NuGet/NuGetGallery#7787]
### Is your feature request related to a problem? Please describe.
Applications that follow catalog are most of the time interested in catalog tail - last few pages to process the changes since the last attempt. To figure out what are the last pages, however, app needs to download the [catalog index](https://api.nuget.org/v3/catalog0/index.json) every time. Currently, this file is 1.7 MB in size (and it only grows), changes almost all the time, and is not compressed. So, if app needs to process changes once a minute it has to download it once a minute, generating 100 MB of network traffic per hour while it only needs the last few records from it. This is especially bad if one wants to poll catalog from a web page: it will hog user's network traffic and if there are 100 users, it will be 100 times as bad for the server hosting the blob.

### Existing workarounds
* Since page numbering in nuget.org is pretty straightforward, instead of checking the catalog index for new pages, the job can probe for the next page by itself. This is really implementation specific and can be easily be broken;
* Range requests on the catalog index. Relies on the service hosting it to support them, on the catalog index to be append only and never change in the middle, which is not really true: ![image](https://user-images.githubusercontent.com/102933/71312909-f13cd480-23e5-11ea-9448-77e3abb910ac.png) It got better recently, but still, it's not documented that there is any order to the catalog index items, also implementation would require to be able to parse partial JSON.

#### Catalog tail resource
Sounds like the simplest thing to do: introduce a new resource that will contain last few entries (last 10, last week, does not matter much) from the catalog index. This will provide the required information for the jobs in most cases using the fraction of network bandwidth compared.

A db2catalog modification would be needed to drop a blob with the last few entries of the catalog index whenever it gets updated provided at best effort: db2catalog will write the catalog index first, if that succeeds, will write the tail, if that fails it will not retry (assuming the blob itself was not corrupted during the attempt, it will be brought up-to-date state at next catalog update); if catalog index update fails, no tail write attempt is going to be made.

Proposed shape:
```JSON
{
  "commitId": "da5e4a25-eb45-4fc1-9522-e5363940db83",
  "commitTimeStamp": "2019-12-21T19:31:28.1771475Z",
  "count": 9380,
  "nuget:lastCreated": "2019-12-21T19:31:26.053Z",
  "nuget:lastDeleted": "2019-12-20T15:40:41.667804Z",
  "nuget:lastEdited": "2019-12-21T19:31:26.053Z",
  "latestItems": [
    {
      "@id": "https://api.nuget.org/v3/catalog0/page9377.json",
      "@type": "CatalogPage",
      "commitId": "9e348716-dbba-4322-bcfa-2c8a59102687",
      "commitTimeStamp": "2019-12-21T07:19:46.8969541Z",
      "count": 549
    },
    {
      "@id": "https://api.nuget.org/v3/catalog0/page9378.json",
      "@type": "CatalogPage",
      "commitId": "02fa4280-e93d-4cd7-baba-a82f3ae8b92c",
      "commitTimeStamp": "2019-12-21T17:03:22.1844852Z",
      "count": 550
    },
    {
      "@id": "https://api.nuget.org/v3/catalog0/page9379.json",
      "@type": "CatalogPage",
      "commitId": "da5e4a25-eb45-4fc1-9522-e5363940db83",
      "commitTimeStamp": "2019-12-21T19:31:28.1771475Z",
      "count": 65
    }  
  ]
}
```
I.e. it will have the same shape as the catalog index (I don't have an opinion on `@id`, `@type` and `@context` properties, can keep them for consistency) with the exception of the `items` property replaced with the `latestItems`. I am a bit unsure about the property rename, though: if the name is kept the same, apps would be able to effortlessly fall back to the catalog index if there are issues with the. 

The `commitId`, `commitTimeStamp` and others are nice to have and provide a way to validate that tail is not lagging behind the main index too much.

#### Organize the catalog pages into a linked list

Change the protocol to make page blobs to include the link to the next page if there is one.

Proposed shape:

```JSON
{
  "@id": "https://api.nuget.org/v3/catalog0/page3615.json",
  "@type": "CatalogPage",
  "commitId": "76181692-ddf1-4d03-b6b9-0cc1046b30e7",
  "commitTimeStamp": "2018-03-30T12:45:42.3023362Z",
  "count": 550,
  "parent": "https://api.nuget.org/v3/catalog0/index.json",
  "nextSibling": "https://api.nuget.org/v3/catalog0/page3616.json", // this
  "items": [ ... ]
}
```

I.e. `nextSibling` (mostly to align with [DOM API](https://developer.mozilla.org/en-US/docs/Web/API/Node/nextSibling), alternatively can be `nextPage`, `nextChild` or whatever else) property is added to a page and updated any time a page after the current is added.

### Additional context
This all hinges on the assumption that there is an interest for processing the catalog by third parties. If there is little demand, probably not worth implementing.
